include/seqan3/core/simd/view_to_simd.hpp

```diff
@@ -532,9 +539,16 @@ class view_to_simd<urng_t, simd_t>::iterator_type
             // Subtract one from last_chunk_pos to get the correct 0-based index.
             final_chunk_pos = std::max<int8_t>(final_chunk_pos, --last_chunk_pos);
         }
+
         final_chunk = all_iterators_reached_sentinel();
-        simd::transpose(matrix);
 
+        if (final_chunk)
+        { // Store the size of the last chunk.
+            size_t size_of_last_chunk = max_size_of_last_chunk % chunk_size;
```

Comment:
`max_size_of_last_chunk` is misleading, because it can be bigger than chunk_size. Or are you referring to a different chunk? If think `max_remaining_elements` would be better.